### DISCUSSION TITLE
Error [ERR_MODULE_NOT_FOUND]

### DISCUSSION THREAD

**Discussion by Nacholoop** [Ride-The-Lightning/RTL#1215] [Troubleshooting Questions]
Hi, anyone can help me??? i cant install RTL, always the same error.... Thank you!!

node rtl
node:internal/errors:490
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/home/xxx/RTL/node_modules/request-promise/' imported from /home/xxx/RTL/backend/utils/common.js
    at new NodeError (node:internal/errors:399:5)
    at legacyMainResolve (node:internal/modules/esm/resolve:235:9)
    at packageResolve (node:internal/modules/esm/resolve:876:14)
    at moduleResolve (node:internal/modules/esm/resolve:938:20)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.15.0

**Comment by ShahanaFarooqui** [comment]
Though this error should not be thrown if you install dependencies with ‘npm install —legacy-peer-deps —omit=dev’ but you can now remove it by installing it separately with ‘npm install request-promise’

**Reply by Nacholoop** [reply]
Thanks for your answer, I've tried both of them but it doesn´t work...

**Comment by ShahanaFarooqui** [comment]
Same error even after installing it explicitly? Need more information to be able to answer.

**Reply by Nacholoop** [reply]
with: npm install —legacy-peer-deps —omit=dev
this result:

npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.

changed 1 package, and audited 362 packages in 3s

25 packages are looking for funding
  run `npm fund` for details

1 moderate severity vulnerability

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

with: npm install request-promise

this error:

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @angular-devkit/build-angular@15.2.4
npm ERR! Found: protractor@3.3.0
npm ERR! node_modules/protractor
npm ERR!   dev protractor@"^3.3.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional protractor@"^7.0.0" from @angular-devkit/build-angular@15.2.4
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"^15.0.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: protractor@7.0.0
npm ERR! node_modules/protractor
npm ERR!   peerOptional protractor@"^7.0.0" from @angular-devkit/build-angular@15.2.4
npm ERR!   node_modules/@angular-devkit/build-angular
npm ERR!     dev @angular-devkit/build-angular@"^15.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

I`ve tried with "npm install request-promise --force" too, with the same result.

**Reply by ShahanaFarooqui** [reply]
The command should be `npm install --legacy-peer-deps --omit=dev` with 2 `-` signs.

**Reply by ShahanaFarooqui** [reply]
My past reply was from my iphone and it replace -- with —. Also I would suggest to delete node_modules and start a fresh installation at this point.
